storage/jellyfish_merkle/src/restore/mod.rs

```diff
@@ -344,6 +293,77 @@ where
         Ok(())
     }
 
+    /// Inserts a new account at the position of the existing leaf node. We may need to create
+    /// multiple internal nodes depending on the length of the common prefix of the existing key
+    /// and the new key.
+    fn insert_at_leaf<'b>(
+        &mut self,
+        child_index: usize,
+        existing_leaf: LeafNode,
+        new_key: HashValue,
+        new_value: AccountStateBlob,
+        mut remaining_nibbles: NibbleIterator<'b>,
```

Comment:
Would it be easier to just pass the nibble path and take `.nibbles()` inside this function?  Then you shouldn't have to bother with lifetimes.  It also seems cleaner to me to just pass the path rather than an iterator.